src/core/imap.ts

```diff
@@ -83,6 +91,81 @@ export class IMAP {
         });
     }
 
+    fetchAll(messages: number[], structure: object) {
+        return new Promise<string[]>((resolve, reject) => {
+            const fetch = this.imap.fetch(messages, structure);
+            const convertedMessages: string[] = [];
+
+            // For each message
+            fetch.on('message', (message, seqno) => {
+                let data = '';
+
+                message.on('body', (stream) => {
+                    stream.on('data', function(chunk) {
+                        // push data
+                        data += chunk;
+                    });
+                });
+
+                message.once('end', async () => {
+                    convertedMessages.push(data);
+                });
+            });
+
+            // All messages have been fetched of type RFC 822.
+            fetch.on('end', () => {
+                resolve(convertedMessages);
+            });
+
+            this.imap.once('error', function(err: Error) {
+                reject(err);
+            });
+        });
+    }
+
+    newMessages(messages: number[]): Promise<number[]> {
+        return new Promise(async (resolve, reject)=>{
+            if (messages.length === 0) {
+                return resolve([]);
+            }
+
+            try {
+                const value = await joplin.settings.value(CONVERTED_MESSAGES);
```

Comment:
This setting will grow over time, could become a performance bottleneck at some point.

Reply:
I came up with a better solution: get the converted message and store it in a variable, then query that variable to see if there is a new message, and when there is, the values are updated. This means that the converted message in Joplin is not only updated until there is a new message(If you have another solution please tell me).

Reply:
Not sure what you mean by variable here? You need to persists this information between restarts somehow.

It seems to me that the easiest way to store this information is to use the email account itself - you either delete messages you've converted, mark them read, or move to another folder.
Or you could only store the date of the last seen email and only process newer. This way you store it on the client but at least it's just one number per mailbox.

Reply:
indeed, the thought came to my mind to store this information by using the email account itself, but there is no standard way to mark the message. To be more clear, some email providers allow you to mark messages while others don't.

But I found a solution One of the flags that are standard for all email providers is the '**SEEN**' flag. Every message is flagged as '**UNSEEN**' if the message is new or the user marks it as '**UNSEEN**'.

So I will add an additional condition in the search, which is to search for the messages that are "from" a specific email and marked as "**UNSEEN**" and then mark the messages as "**SEEN**" after fetching them.

I think this solution is better than deleting. and the user can mark as an "**unread**" message any number of messages in their email provider.

Sorry for the delay in suggesting this solution. It took me a lot of time to test this solution across different email providers.

Reply:
example:

https://user-images.githubusercontent.com/58605547/185809305-565da1bb-046c-46cf-975e-45d0fcc75015.mp4

Reply:
Just to confirm - you'll add this in another pr, right?